qiime2/sdk/util.py

```diff
@@ -118,3 +120,9 @@ def actions_by_input_type(string):
                 commands.append((pg, actions))
 
     return commands
+
+
+def validate_result_collection_key(key):
```

Comment:
I recommend adding a couple of unit tests for this function itself, especially since it's public facing the in sdk.